test/libsolidity/syntaxTests/errors/error_as_function_param.sol

```diff
@@ -1,4 +1,4 @@
 error E(uint);
 function f(E x) pure returns (uint) {}
 // ----
-// TypeError 5172: (26-27): Name has to refer to a struct, enum or contract.
+// TypeError 5172: (26-27): Name has to refer to a user-defined type.
```

Comment:
Yeah, looking at it here, I wish it were a little more 'explicit', although explicit may be the wrong word for it, since it's already explicit, as long as you know the exact definition of user defined types in the context of Solidity :D. Perhaps having something like `(e.g. structs, enums, bla bla)` after it?

Reply:
Not sure. I think I'd rather keep it generic because otherwise we should really include that in every error message that mentions them.

I think we should instead spend more effort making sure these terms are defined in the docs and used consistently. That's actually something that's really bugging me in the operator PR because we don't use a single, unambiguous term to refer to them. Instead we have a mix of "custom operators", "user-defined operators", "user type operators" and I think there were even a few more variants before I corrected them in the PR. Having a cleanup of the terminology used in docs, names and messages is one thing I'd like to as a follow up to the operator PR. Maybe we should do that for "user-defined type" too.